pkg/frontend/admin_openshiftcluster_resources_list.go

```diff
@@ -0,0 +1,58 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/resources"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func (f *frontend) listAdminOpenShiftClusterResources(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+
+	jb, err := f._listAdminOpenShiftClusterResources(ctx, r)
+
+	reply(log, w, nil, jb, err)
+}
+
+func (f *frontend) _listAdminOpenShiftClusterResources(ctx context.Context, r *http.Request) ([]byte, error) {
+	vars := mux.Vars(r)
+	resourceID := strings.TrimPrefix(r.URL.Path, "/admin")
+
+	doc, err := f.db.OpenShiftClusters.Get(ctx, resourceID)
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		return nil, api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s' under resource group '%s' was not found.", vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
+	case err != nil:
+		return nil, err
+	}
+
+	fpAuthorizer, err := f.env.FPAuthorizer(f.env.TenantID(), azure.PublicCloud.ResourceManagerEndpoint)
```

Comment:
@jim-minter I will move this, I am just not sure on what authorizer to use.
This is currently not returning what i expect, it returns the resources, but with very limited info, but if do
```az resource list -g v4-angus-rg -o table --debug```
then each resource is properly populated, I am guessing that it is the difference in auth? - as the sent url is the same in both cases.

Reply:
I believe this is correct.

Reply:
I don't think there's an auth difference.  I think you would need to iterate the resources returned by list, then get each individually.  You should be able to use some resources client to do that (maybe even the one you've got).  It's a bit like the Azure equivalent of the k8s dynamic API.

Reply:
Thanks @jim-minter this works nicely now.